doc/sphinx/addendum/implicit-coercions.rst

```diff
@@ -167,25 +167,38 @@ Declaring Coercions
      Silence the non uniform inheritance warning.
 
   .. exn:: @qualid not declared.
-     :undocumented:
+
+     The given :token:`qualid` is not a defined constant.
```

Comment:
```suggestion
     :token:`qualid` is not a defined constant.
```

Reply:
Note that "defined" is sightly misleading, as I don't think `Axiom foo` defines `foo`

Reply:
What would I say instead? `@qualid does not appear in the global context`?

Reply:
Sure.  Though that's ambiguous because I believe coercions in sections can target context variables, so it's not clear that "global" is quite right.  (Am I misremembering?  This should probably be explicit in the doc if it's not already, in any case.). Maybe `@qualid is not a defined constant, (co)inductive, axiom, etc`?  @jfehrle does the document have a standard way of referring to the concept of "an identifier which is present in the global / section environment"?

Reply:
OK I've gone with `@qualid is not a defined constant, (co)inductive, axiom, etc` but @jfehrle can enlighten us if global context makes sense to him.

Reply:
Why not make it just "@qualid is not defined"?  The "etc." is a bit vague.  If there are specific categories that are excluded, you might mention them in the description.

> Note that "defined" is slightly misleading, as I don't think `Axiom foo` defines `foo`

Using the everyday meaning of "define", I think users expect that `Axiom foo` does indeed define something.  Giving "define" a technical meaning will be a bit confusing and would mean rewording/clarifying some of the uses of  "define", which appears more than 500 times in the doc.  I would not go there today.

FWIW, the doc defines "global environment" but not "global context".

Reply:
>Why not make it just "@qualid is not defined"? The "etc." is a bit vague. If there are specific categories that are excluded, you might mention them in the description.

I think I originally didn't want to give the impression that it had to be a definition.

>FWIW, the doc defines "global environment" but not "global context".

Yes, that is more correct although they could be used interchangeably in academic vernacular not specific to Coq.

I also thought saying `@qualid is not defined` is kind of useless since the error message already has this information.

Reply:
> I also thought saying `@qualid is not defined` is kind of useless since the error message already has this information.

If the message is self explanitory, keep the `:undocumented:`

Reply:
(I am happy to defer to Jim on wording here)

Reply:
This was the original reason I started this PR actually. I still think adding to `  .. exn:: @qualid not declared.` is important since I was confused before by something being in the local environment and not being picked up by coercions, it was only when I understood it being in the global environment that it made sense.

Reply:
I've changed it it to `not defined globally`.